drivers/hyperv/hyperv.go

```diff
@@ -447,7 +475,33 @@ func (d *Driver) GetIP() (string, error) {
 		return "", fmt.Errorf("IP not found")
 	}
 
-	return resp[0], nil
+	switch d.PreferredNetworkProtocol {
+	case PreferIPv4:
+		for _, ipStr := range resp {
+			ip := net.ParseIP(ipStr)
+			if isIPv4(ipStr) && ip.To4() != nil && ip.IsGlobalUnicast() {
```

Comment:
See previous comment... it would make this check easier as no To4 is needed, as we already know this.